### DISCUSSION TITLE
How to simplify the environment readiness for Run Test Cases

### DISCUSSION THREAD

**Discussion by lewiseliu** [apache/apisix#7741] [Q&A]
Hi Community, 

**Background**
Development environment buildup should include 2 parts:

- part 1: APISIX Installation and management, which was already clarified in Discussion #7705 ([How to speed up APISIX Development environment buildup #7705 ), and we shared Docker image for community reference. 
- part 2: Run Test Cases, we are following the guidance on step 4  - Run Test Cases of https://apisix.apache.org/docs/apisix/2.13/how-to-build/. which is blocked and great helps are expected from Community here. 

**Detail Issues:**
After installing cpanm and test-nginx(perl Makefile.PL), prove test can not work well. 
 
**Help expected:**
1. Any more detail guidance can be shared, since the currrent official one is more simple and many manual steps need to do, eg. yum install was not mentioned, include yum install perl-Test-Harness perl-ExtUtils-MakeMaker perl-CPAN  perl-Digest-MD5 -y 
2. How to fix module 'resty.ipmatcher' not found

**Detail steps:-**

- Yum install: yum install perl-Test-Harness perl-ExtUtils-MakeMaker perl-CPAN  perl-Digest-MD5 -y
- Install cpanminus, the package manager for perl. (PASSED)
- Install the test-nginx dependencies via cpanm as shown below. sudo cpanm --notest Test::Nginx IPC::Run > build.log 2>&1 || (cat build.log && exit 1)   (PASSED)
- Clone the latest source code locally by using the forked out version. git clone https://github.com/openresty/test-nginx.git, e.g. git clone and perl Makefile.PL   (PASSED)
- Run the tests. Opt 1: Append the current directory to the perl module directory: export PERL5LIB=.:$PERL5LIB and then run make test command. (PASSED);  
Opt 2: Specify the NGINX binary path by running PATH=/usr/local/openresty/nginx/sbin:/usr/bin PERL5LIB=.:$PERL5LIB FLUSH_ETCD=1 prove -Itest-nginx/lib -r t/admin  (FAILED)

Logs:
Opt 2: prove test
PATH=/usr/local/openresty/nginx/sbin:/usr/bin PERL5LIB=.:$PERL5LIB FLUSH_ETCD=1 prove -Itest-nginx/lib -r t/admin
`t/admin/api.t .................... nginx: [error] init_by_lua error: /data/apisix.swissre/apisix/apisix/patch.lua:19: module 'resty.ipmatcher' not found`

**Comment by juststillthinking** [comment]
The best way to run test case is  follow the ci file: https://github.com/apache/apisix/blob/master/ci/centos7-ci.sh

**Reply by tzssangglass** [reply]
maybe you miss the step of `create_lua_deps(make deps)`, ref: https://github.com/apache/apisix/blob/308bdd967792bb7b34355b98dcf5d6cdeae078f6/ci/centos7-ci.sh#L80

**Reply by lewiseliu** [reply]
@monkeyDluffy6017  @tzssangglass  When we run the ci/centos7-ci.sh, in below step some error is raised as below. Any idea, thank you
```
# yum install -y openresty openresty-debug openresty-openssl111-debug-devel pcre pcre-devel
.............
Error: apisix-base conflicts with openresty-1.21.4.1-1.el7.x86_64
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest
``` 
The fully logs as here.
[centos7-ci.log](https://github.com/apache/apisix/files/9388250/centos7-ci.log)

**Reply by juststillthinking** [reply]
Where does the `apisix-base` come from ? Have you install apisix before with yum ? test env should work with the apisix source code

**Reply by tzssangglass** [reply]
you only need to install `apisix-base`, uit is an OpenResty distribution of APISIX. Don't need to install `openresty`
